x-pack/plugins/actions/server/saved_objects/transform_connectors_for_export.ts

```diff
@@ -31,12 +31,34 @@ function transformConnectorForExport(
   try {
     // If connector requires secrets, this will throw an error
     validateSecrets(actionType, {});
+  } catch (err) {
```

Comment:
What do you think of simplifying this a little so we're not validating the secrets twice?

I think this might work (tried it quickly but haven't tested thoroughly):
```
try {
    // If connector requires secrets, this will throw an error
    validateSecrets(actionType, {});

    // this is the new code block
    const hasAuth = connector?.attributes?.config?.hasAuth;

    // If connector has optional (or no) secrets, set isMissingSecrets value to value of hasAuth
    // If connector doesn't have hasAuth value, default to isMissingSecrets: false
    isMissingSecrets = (hasAuth as boolean) ?? false;

    // if hasAuth is strictly equal to 'false', then we know it's defined and set to false and we should set secrets to 
   // {}
    if (hasAuth === false) {
      return {
        ...connector,
        attributes: {
          ...connector.attributes,
          secrets: {},
          isMissingSecrets,
        },
      };
    }

// everything after this is the same
  } catch (err) {
    isMissingSecrets = true;
  }
```

Reply:
Sure, I think that will work! I will update it

Reply:
Is `hasAuth` not null for connectors where secrets are optional or required?

Reply:
currently `hasAuth` is not standard across all connectors, so only email & webhook connectors use it. It should be `undefined` for all other connector types.

Reply:
okay great, thanks!

Reply:
Resolved in this [commit](https://github.com/elastic/kibana/pull/138019/commits/805d9825f7a051f4e3daa7157deefa4588332821)